kernel/common/numeric.rb

```diff
@@ -31,23 +31,27 @@ def <=>(other)
     return nil
   end
 
-  def step(limit, step=1)
+  def step(limit=nil, step=nil, to: nil, by: nil)
     unless block_given?
-      return to_enum(:step, limit, step) do
-        Rubinius::Mirror::Numeric.reflect(self).step_size(limit, step)
+      return to_enum(:step, limit, step, to: to, by: by) do
+        Rubinius::Mirror::Numeric.reflect(self).step_size(limit, step, to, by)
       end
     end
 
-    raise ArgumentError, "step cannot be 0" if step == 0
-
     m = Rubinius::Mirror::Numeric.reflect(self)
-    values = m.step_fetch_args(limit, step)
+    values = m.step_fetch_args(limit, step, to, by)
     value = values[0]
     limit = values[1]
     step = values[2]
     asc = values[3]
     is_float = values[4]
 
+    if(step == 0)
+      while true
+        yield self
```

Comment:
How is this supposed to ever return from the method (other than the supplied block calling `break`)?